File: app/src/main/java/com/jorotayo/algorubickrevamped/data/Category.java
package com.jorotayo.algorubickrevamped.data;

import io.objectbox.annotation.Entity;
import io.objectbox.annotation.Id;

@Entity
public class Category {

    @Id
    public long id;
    public String category_name;

    public Category(String category_name) {
        this.category_name = category_name;
    }

    public Category(){

    }

    public String getCategory_name() {
        return category_name;
    }

    public void setCategory_name(String name) {
        this.category_name = name;
    }
}

File: app/src/main/java/com/jorotayo/algorubickrevamped/data/Category.kt
package com.jorotayo.algorubickrevamped.data

import io.objectbox.annotation.Entity
import io.objectbox.annotation.Id

@Entity
data class Category(
    @Id var id: Long = 0,
    var category_name: String = "",
) {
    constructor(category_name: String) : this(id = 0, category_name = category_name)
}
